### PR TITLE
feat(metrics): update metrics for request and operation latency

### PR DESCRIPTION
**Pull request by pimeys**
The last pieces in grafbase/grafbase. This PR will break the API and our own analytics, so we need to fix that before merging.

### CODE REVIEW

File: engine/crates/telemetry/src/metrics/request.rs [hunk 1]
```diff
@@ -35,23 +42,53 @@ impl RequestMetrics {
...
+            attributes.push(KeyValue::new("server.port", listen_address.port() as i64));
+        }
+
+        if let Some(scheme) = url_scheme {
+            attributes.push(KeyValue::new("url.scheme", scheme.to_string()));
+        }
+
         if let Some(cache_status) = cache_status {
-            attributes.push(KeyValue::new("http.response.headers.cache_status", cache_status));
+            attributes.push(KeyValue::new("http.response.headers.cache.status", cache_status));
```
Comment (Finistere) [inline engine/crates/telemetry/src/metrics/request.rs:77]: maybe `x-grafbase-cache`?

File: engine/crates/telemetry/src/metrics/request.rs [hunk 2]
```diff
@@ -17,12 +19,17 @@ pub struct RequestMetricsAttributes {
     pub cache_status: Option<String>,
     pub gql_status: Option<GraphqlResponseStatus>,
     pub client: Option<Client>,
+    pub url_scheme: Option<String>,
+    pub route: Option<String>,
```
Comment (Finistere) [inline engine/crates/telemetry/src/metrics/request.rs:23]: Think could be a `&'static str`. Not sure we'll ever need a dynamic string even in the future

**Review by Finistere — APPROVED** [review]

File: engine/crates/telemetry/src/metrics/request.rs (hunk 2 above)
Comment (pimeys) [reply engine/crates/telemetry/src/metrics/request.rs:23]: Ummm... we do. https://grafbase.com/docs/self-hosted-gateway#graph-configuration

Comment (Finistere) [reply engine/crates/telemetry/src/metrics/request.rs:23]: hum... ok fair point forgot that was configurable. Could `Box::leak` I guess but with config reloading and all...

**Pull request merged**